apis/datadoghq/v1alpha1/datadogagentprofile_validation.go

```diff
@@ -0,0 +1,45 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package v1alpha1
+
+import (
+	"fmt"
+)
+
+// IsValidDatadogAgentProfile is used to check if a DatadogAgentProfileSpec is valid
+func IsValidDatadogAgentProfile(spec *DatadogAgentProfileSpec) error {
+	// check that profileAffinity contains a set of requirements
+	if spec.ProfileAffinity == nil {
+		return fmt.Errorf("profileAffinity must be defined")
+	}
+	if spec.ProfileAffinity.ProfileNodeAffinity == nil {
+		return fmt.Errorf("profileNodeAffinity must be defined")
+	}
+	if len(spec.ProfileAffinity.ProfileNodeAffinity) < 1 {
+		return fmt.Errorf("profileNodeAffinity must have at least 1 requirement")
+	}
+
+	// validate config
+	if spec.Config == nil {
+		return fmt.Errorf("config must be defined")
+	}
+	if spec.Config.Override == nil {
+		return fmt.Errorf("config override must be defined")
+	}
+	if spec.Config.Override[NodeAgentComponentName] == nil {
+		return fmt.Errorf("node agent override must be defined")
+	}
+	if spec.Config.Override[NodeAgentComponentName].Containers == nil {
+		return fmt.Errorf("node agent container must be defined")
+	}
+	for name, container := range spec.Config.Override[NodeAgentComponentName].Containers {
+		if container.Resources == nil {
```

Comment:
I don't think this is right because it means that the profile needs an override for each of the containers (core agent, trace agent, etc.).

I think that we should only require one. In the future, when we add support for other things, like enabling/disabling features, not even one resources override should be required, because it should be possible to have profiles that enable/disable features without specifying any resource limits/requests.